### PR TITLE
make use of Reflectable

### PR DESCRIPTION
**Pull request by valotas**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,13 +125,41 @@ pub global run coverage:collect_coverage --uri=http://... -o /tmp/mustache4dart.
...
+[`reflectable`](https://pub.dartlang.org/packages/reflectable) is being used.
+In such a case the user should mark the objects used by mustache4dart with
+`@MustacheContext()` and make sure that the reflectable builder is being run
+accordingly.
+
+### `reflectable` setup
+
+The easiest way to get reflectable to work is by making use of
+[`build_runner`](https://pub.dartlang.org/packages/build_runner) with a
+`build.yalm` that should look like:
```
Comment (srawlins) [inline README.md]: typo: `build.yaml`.

File: README.md [hunk 2]
```diff
@@ -122,13 +125,41 @@ pub global run coverage:collect_coverage --uri=http://... -o /tmp/mustache4dart.
...
+
+```
```
Comment (srawlins) [inline README.md]: Can give the language here with <code>```yaml</code>.

**Comment by srawlins** [comment]
Woohoo excited about this! Thanks @valotas !

File: README.md [hunk 3]
```diff
@@ -74,16 +77,44 @@ print(salut({'name': 'Alice'})); //should print Hello Alice!
...
-----------
-The project passes all the [Mustache specs][specs].  You have to make sure
+The easiest way to get reflectable to work is by making use of
+[`build_runner`](https://pub.dartlang.org/packages/build_runner) with a
+`build.yaml` that should look like:
+
+```yaml
+targets:
+  # your package name, in my case is mustache4dart
+  mustache4dart:
```
Comment (jakemac53) [inline README.md:104]: Note that you can do `$default` here which works for any app (it fills in the package name).

**Pull request merged**